docs/data/base/components/transitions/AllTransitionsDemo.tsx

```diff
@@ -0,0 +1,202 @@
+import * as React from 'react';
+import { styled } from '@mui/system';
+import { Unstable_Popup as BasePopup, PopupProps } from '@mui/base/Unstable_Popup';
+import { Button as BaseButton } from '@mui/base/Button';
+import { CssAnimation, CssTransition } from '@mui/base/Transitions';
+
+const styles = `
+  @keyframes open-animation {
+    0% {
+      opacity: 0;
+      transform: translateY(-8px) scale(0.95);
+    }
+
+    50% {
+      opacity: 1;
+      transform: translateY(4px) scale(1.05);
+    }
+
+    100% {
+      opacity: 1;
+      transform: translateY(0) scale(1);
+    }
+  }
+
+  @keyframes close-animation {
+    0% {
+      opacity: 1;
+      transform: scale(1) rotate(0deg);
+      filter: blur(0);
+    }
+
+    100% {
+      opacity: 0;
+      transform: scale(1.5) rotate(8deg);
+      filter: blur(4px);
+    }
+  }
+
+  .anim-open {
+    animation: open-animation 1s ease-out both;
+  }
+
+  .anim-close {
+    animation: close-animation 1s ease-out forwards;
+  }
+
+  .open {
+    opacity: 1;
+    transform: translateY(0) scale(1)
+    filter: blur(0);
+    transition: transform 0.2s cubic-bezier(0.345, 0.275, 0.505, 1.625), opacity 0.2s ease-out;
+  }
+
+  .close {
+    opacity: 0;
+    transform: translateY(-8px) scale(0.95);
+    filter: blur(3px);
+    transition: transform 0.4s ease-out, opacity 0.4s ease-out, filter 0.2s ease-out;
+  }
+`;
+
+const grey = {
+  200: '#DAE2ED',
+  700: '#434D5B',
+  900: '#1C2025',
+};
+
+const blue = {
+  200: '#99CCFF',
+  300: '#66B2FF',
+  400: '#3399FF',
+  500: '#007FFF',
+  600: '#0072E5',
+  700: '#0066CC',
+};
+
+const Button = styled(BaseButton)(
+  ({ theme }) => `
+  min-width: 165px;
+  font-family: 'IBM Plex Sans', sans-serif;
+  font-weight: 600;
+  font-size: 0.875rem;
+  line-height: 1.5;
+  background-color: ${blue[500]};
+  padding: 8px 16px;
+  border-radius: 8px;
+  color: white;
+  transition: background-color, box-shadow, 120ms ease;
+  cursor: pointer;
+  border: 1px solid ${blue[500]};
+  box-shadow: 0 1px 1px ${
+    theme.palette.mode === 'dark' ? 'rgba(0, 0, 0, 0.5)' : 'rgba(45, 45, 60, 0.1)'
+  }, inset 0 1.5px 1px ${blue[400]}, inset 0 -2px 1px ${blue[600]};
+
+  &:hover {
+    background-color: ${blue[600]};
+  }
+
+  &:active {
+    background-color: ${blue[700]};
+    box-shadow: none;
+  }
+
+  &:focus-visible {
+    box-shadow: 0 0 0 4px ${theme.palette.mode === 'dark' ? blue[300] : blue[200]};
+    outline: none;
+  }
+`,
+);
+
+const StyledPopup = styled(BasePopup)`
```

Comment:
Would make more sense to me

```suggestion
const Popup = styled(PopupBase)`
```